pkg/csi/cinder/nodeserver.go

```diff
@@ -491,10 +492,17 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 		return nil, status.Error(codes.Internal, "Unable to find Device path for volume")
 	}
 
+	// comparing current volume size with the expected one
+	newSize := req.GetCapacityRange().GetRequiredBytes()
+	if err := blockdevice.PollBlockGeometry(devicePath, volumePath, newSize); err != nil {
```

Comment:
@kayrus Im not very comfortable issuing this for every call , when it is needed only for VMware Hypervisor. It is unnecessary if size is already updated , Is there a check that can be done to only to rescan if size is not updated or so to avoid overhead.  May be we could take others opinion on this . @lingxiankong @adisky

Reply:
@ramineni +1

Reply:
@ramineni @adisky not it polls geometry only, when the size is wrong